### ISSUE TITLE
No Open in browser

### ISSUE THREAD

**Issue by meetgyn** [softrams/bulwark#459]
I did all the procedure described in the post, but when trying to access it through the browser, it does not open.
localhost: 5000, nothing happens.

**Comment by Whamo12** [comment]
@meetgyn Thanks for reaching out.  Are you setting up the application via the Docker or locally?

**Comment by meetgyn** [comment]
Yes, I am using a docker. It didn't show any error, however when trying to access it through the browser, I only get time out, and it doesn't open. I made several configurations here but they were successful.

**Comment by Whamo12** [comment]
Thanks for the information. Do you see any errors in the web browser console or in the docker terminal? 

cc @skewled

**Comment by meetgyn** [comment]
No error is returned, in the docker terminal, or web browser.
It just doesn't open. I tested with the ip http://172.16.16.3 anymore
http: // localhost: 5000
http://192.168.15.20:53000
but nothing happens. I find it strange because I have some containers in my docker and they are all working perfectly.

**Comment by skewled** [comment]
@meetgyn please perform the following.

In your preferred terminal:

- `docker-compose down`
- `docker-compose up`

Submit screenshots from all command output into this issue please.

**Comment by meetgyn** [comment]
![image](https://user-images.githubusercontent.com/3695608/100162786-29208780-2e93-11eb-9863-b6f109f00143.png)


![image](https://user-images.githubusercontent.com/3695608/100162486-86680900-2e92-11eb-9818-8f25293be336.png)


![image](https://user-images.githubusercontent.com/3695608/100162617-cb8c3b00-2e92-11eb-8a74-5fca2b52292c.png)

![image](https://user-images.githubusercontent.com/3695608/100162656-e3fc5580-2e92-11eb-91a3-488b1ac1d2bc.png)

**Comment by skewled** [comment]
@meetgyn Thank you, and it appears mysql is not launching properly or listening for communications. We've found a comon culprit with this issue is due to the `.env` file not being created properly. Please perform the below and let us know the outcome:

- Cleanup -

- `press ctrl+c to abort the launch`
- `docker-compose down`
- `docker volume rm bulwark_bulwark-db`
- `docker volume rm bulwark_bulwark-temp`

- Setup -

- Ensure you have the correct file created `.env` without any filetype tag ex: text, log, docx. **The `.` in front of this file is also very important.
- Populate with default values provided in the README file in the repository source.

- Startup -

- `docker-compose up` - wait for signs of life, with any luck the `.env` file will be processed and you'll be up and running just fine.

**Comment by meetgyn** [comment]
I created the .env file in the terminal: touch .env
I did all the procedure that passed me, but this is still with mysql error

![image](https://user-images.githubusercontent.com/3695608/100164635-0c864e80-2e97-11eb-929e-87bb10c4ab7a.png)

**Comment by skewled** [comment]
The `"changeme"` portions are up to you for random values and you should keep them private once set.

It looks like you're using Windows 7 with Putty to connect to a Linux Host.

What Linux Distribution and Version?
What version of docker?
What version of docker-compose?

**Comment by meetgyn** [comment]
![image](https://user-images.githubusercontent.com/3695608/100166149-c632ee80-2e9a-11eb-9047-00e804b1f7f9.png)
![image](https://user-images.githubusercontent.com/3695608/100166191-e06ccc80-2e9a-11eb-847c-1fed7e227e63.png)

**Comment by skewled** [comment]
I setup a Debian 10 - Buster Virtual Machine
Installed docker according to the docker documentation.
Installed docker-compose according to the docker documentation.

`
root@debian:~/repos/bulwark# lsb_release -cs
buster
root@debian:~/repos/bulwark# uname -a
Linux debian 4.19.0-12-amd64 #1 SMP Debian 4.19.152-1 (2020-10-18) x86_64 GNU/Linux
root@debian:~/repos/bulwark# docker --version
Docker version 19.03.13, build 4484c46d9d
root@debian:~/repos/bulwark# docker-compose --version
docker-compose version 1.27.4, build 40524192
`

I was able to issue the `docker-compose up` command and download the containers as well as launch Bulwark.
During the process it is normal to observe the connection to mysql isn't available yet; however, after the mysql container is up and running the error goes away and Bulwark continues its self-provisioning process.

Normal startup messaging:

`
bulwark       | ERROR 2002 (HY000): Can't connect to MySQL server on '172.16.16.3' (115)
bulwark       | MySQL is unavailable - sleeping
`

MySQL is now available and you'll observe:

`
bulwark       | user
bulwark       | bulwark
bulwark       | root
bulwark       | mysql.session
bulwark       | mysql.sys
bulwark       | root
bulwark       | MySQL should be up - starting up Bulwark.
bulwark       | Initial DB Creation
bulwark       | npm info it worked if it ends with ok
bulwark       | npm info using npm@6.14.8
bulwark       | npm info using node@v14.9.0
bulwark       | npm info lifecycle bulwark@6.2.4~predocker:check: bulwark@6.2.4
bulwark       | npm info lifecycle bulwark@6.2.4~docker:check: bulwark@6.2.4
bulwark       | 
bulwark       | > bulwark@6.2.4 docker:check /bulwark
`

Shortly after everything scrolls the terminal, you'll get to this point:

`
bulwark       | A .env file was not found. Attempting to fetch values from existing environment variables.
bulwark       | Server running on 127.0.0.1:5000
`

Bulwark will be available at http://targetIP:5000, in my case it was: http://172.16.95.135:5000/#/login

Further notes, ensure there are no special characters entered into the file. I used nano to create the .env file in the root folder of bulwark. Generally this should be the folder where you've cloned down the repository.

Folder I cloned Bulwark into was titled `repos`
This turns into `/repos/bulwark`
So my `.env` file would be created in the `/repos/bulwark/.env` location

Update your docker-compose to latest stable version.

Hope this helps and you are able to explore Bulwark with docker.
